sdk/go/request.go

```diff
@@ -45,57 +45,53 @@ func (r OnlineFeaturesRequest) buildRequest() (*serving.GetOnlineFeaturesRequest
 		}
 	}
 	return &serving.GetOnlineFeaturesRequest{
-		Features:   features,
+		Features:   featureRefs,
 		EntityRows: entityRows,
 	}, nil
 }
 
-// buildFeatures create a slice of FeatureReferences from a slice of "<project>/<feature_name>:<feature-set-version>"
-// It returns an error when the format is invalid
-func buildFeatures(featureReferences []string, defaultProject string) ([]*serving.FeatureReference, error) {
-	var features []*serving.FeatureReference
-
-	for _, featureRef := range featureReferences {
-		var project string
-		var name string
-		var version int
-		var featureSplit []string
-
-		projectSplit := strings.Split(featureRef, "/")
-
-		if len(projectSplit) == 2 {
-			project = projectSplit[0]
-			featureSplit = strings.Split(projectSplit[1], ":")
-		} else if len(projectSplit) == 1 {
-			project = defaultProject
-			featureSplit = strings.Split(projectSplit[0], ":")
-		} else {
-			return nil, fmt.Errorf(ErrInvalidFeatureName, featureRef)
+// Creates a slice of FeatureReferences from string representation in
+// the format featureset:feature.
+// featureRefStrs - string feature references to parse.
+// project - Sets the project in parsed FeatureReferences. Otherwise ""
+// Returns parsed FeatureReferences.
+// It returns an error when the format of the string feature reference is invalid
+func buildFeatureRefs(featureRefStrs []string, project string) ([]*serving.FeatureReference, error) {
+	var featureRefs []*serving.FeatureReference
+
+	for _, featureRefStr := range featureRefStrs {
+		parseStr := strings.TrimSpace(featureRefStr)
+		if len(parseStr) == 0 {
+			return nil, fmt.Errorf(ErrInvalidFeatureRef, featureRefStr)
 		}
 
-		if len(featureSplit) == 2 {
-			name = featureSplit[0]
-			v, err := strconv.Atoi(featureSplit[1])
+		var featureRef serving.FeatureReference
+		if strings.Contains(parseStr, "/") {
+			return nil, fmt.Errorf(ErrInvalidFeatureRef, featureRefStr)
+		}
+		// parse version if specified
+		hasVersion, _ := regexp.MatchString(".*:\\d+", parseStr)
+		if hasVersion {
```

Comment:
Would you mind shining some light on this code? Wouldn't it be easier to just ignore versions and have Core deal with it? So if you set a version it fails at Core, if you don't set it then it succeeds. So upgrading to Feast 0.5 requires you to remove versions.

Reply:
This code is added as a band aid for testing. Post #676 would be removed.